### PR TITLE
Update MySQL Reference Manuals to use 8.0

### PR DESCRIPTION
**Pull request by dveeden**
### What is changed, added or deleted? (Required)

Change links from `https://dev.mysql.com/doc/refman/5.7/...`  to `https://dev.mysql.com/doc/refman/8.0/...`

This PR containts two commits:
- A commit adding a script
- A commit with the changes made by that script

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v7.4 (TiDB 7.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by ran-huang** [comment]
@dveeden Would it be possible to have a PM or someone in a similar role review this PR? I would like clarification on whether TiDB is compatible with MySQL 8.0 or 5.7 or both. Thank you.

File: functions-and-operators/string-functions.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ aliases: ['/docs/dev/functions-and-operators/string-functions/','/docs/dev/refer
 
 # String Functions
 
-TiDB supports most of the [string functions](https://dev.mysql.com/doc/refman/5.7/en/string-functions.html) available in MySQL 5.7, some of the [string functions](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html) available in MySQL 8.0, and some of the [functions](https://docs.oracle.com/en/database/oracle/oracle-database/21/sqlqr/SQL-Functions.html#GUID-93EC62F8-415D-4A7E-B050-5D5B2C127009) available in Oracle 21.
+TiDB supports most of the [string functions](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html) available in MySQL 5.7, some of the [string functions](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html) available in MySQL 8.0, and some of the [functions](https://docs.oracle.com/en/database/oracle/oracle-database/21/sqlqr/SQL-Functions.html#GUID-93EC62F8-415D-4A7E-B050-5D5B2C127009) available in Oracle 21.
```
Comment (mjonss) [inline functions-and-operators/string-functions.md:9]: Should the link and the text be in sync? I.e. should the text say MySQL 8.0 instead of MySQL 5.7?

**Review by mjonss — COMMENTED** [review]
There are a couple of places where the link is updated to 8.0, but the text is still saying 5.7, will this be addressed in a future PR?

File: functions-and-operators/string-functions.md (hunk 1 above)
Comment (dveeden) [reply functions-and-operators/string-functions.md:9]: Yes, for this specific case the link should not be updated.

```suggestion
TiDB supports most of the [string functions](https://dev.mysql.com/doc/refman/5.7/en/string-functions.html) available in MySQL 5.7, some of the [string functions](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html) available in MySQL 8.0, and some of the [functions](https://docs.oracle.com/en/database/oracle/oracle-database/21/sqlqr/SQL-Functions.html#GUID-93EC62F8-415D-4A7E-B050-5D5B2C127009) available in Oracle 21.
```

**Comment by dveeden** [comment]
> There are a couple of places where the link is updated to 8.0, but the text is still saying 5.7, will this be addressed in a future PR?

I've updated the script to exclude the URLs that mention `available in MySQL 5.7` in the text next to it and re-ran the script.

Best to update those together with the text later when we've checked if the text is still valid for MySQL 8.0.

And `mysql-compatibility.md` will also be updated later.

**Review by ran-huang — APPROVED** [review]

**Review by mjonss — APPROVED** [review]

**Review by ran-huang — APPROVED** [review]

**Comment by ran-huang** [comment]
Some links in functions and operators docs are not updated. Are they left on purpose?
- functions-and-operators/bit-functions-and-operators.md
- functions-and-operators/cast-functions-and-operators.md
- functions-and-operators/control-flow-functions.md
- functions-and-operators/date-and-time-functions.md
- functions-and-operators/encryption-and-compression-functions.md
- functions-and-operators/information-functions.md
- functions-and-operators/locking-functions.md
- functions-and-operators/miscellaneous-functions.md
- functions-and-operators/numeric-functions-and-operators.md
- functions-and-operators/string-functions.md

**Review by Oreoxmt — APPROVED** [review]

**Pull request merged**